internal/generator/generator.go

```diff
@@ -0,0 +1,92 @@
+package generator
+
+import (
+	"fmt"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/nestoca/joy-generator/internal/gitrepo"
+	"github.com/nestoca/joy/api/v1alpha1"
+	"github.com/nestoca/joy/pkg/catalog"
+)
+
+type Generator struct {
+	repo *gitrepo.GitRepo
+}
+
+type Result struct {
+	// Release holds the release's values loaded from the yaml file in the catalog
+	Release *v1alpha1.Release `json:"release"`
+
+	// Environment holds the environment info where the release will be deployed. The full spec is not loaded to minimize the payload size
+	Environment *Environment `json:"environment"`
+
+	// RenderedValues is a yaml string that is the Release.spec.values rendered with any templated fields
+	RenderedValues string `json:"templatedValues"`
+}
+
+type Environment struct {
+	Name        string `json:"name"`
+	ClusterName string `json:"clusterName"`
```

Comment:
Let's stick to same simple property names as in the original Joy `Environment` CRD: `cluster`, pretty please? 😉

Reply:
Similar to the above comment, I wanted to make it clear in the template that this is the name of the cluster, since the `destination` field of an ArgoCD can use either the ArgoCD cluster name, or it's IP/Port.

Reply:
Just to understand, why not reference the actual Joy `Environment` struct here?  We never know what going to be in there that will need to be leveraged in generators.  I saw you are doing that to reduce payload size, but that's really not a huge object anyway.  It would also simplify everybody's understanding and expectations, knowing it's just the Joy `Environment` CRD that's made accessible there. I for one would prefer to keep that assumption in mind when configuring my generator to use that output.

Reply:
Ok i'll add the full spec